packages/svelte2tsx/src/svelte2tsx.ts

```diff
@@ -777,10 +823,34 @@ function addComponentExport(
                 ? '__sveltets_with_any(render().props)'
                 : 'render().props'
             : `__sveltets_partial${uses$$propsOr$$restProps ? '_with_any' : ''}(render().props)`;
-    str.append(
-        // eslint-disable-next-line max-len
-        `\n\nexport default class {\n    $$prop_def = ${propDef}\n    $$slot_def = render().slots\n}`,
-    );
+
+    const doc = formatComponentDocumentation(componentDocumentation);
+
+    // eslint-disable-next-line max-len
+    const statement = `\n\n${doc}export default class ${className ? `${className} ` : ''}{\n    $$prop_def = ${propDef}\n    $$slot_def = render().slots\n}`;
+
+    str.append(statement);
+}
+
+/**
+ * Returns a Svelte-compatible component name from a filename. Svelte
+ * components must use capitalized tags, so we try to transform the filename.
+ *
+ * Works with most filename styles except for filenames with multiple
+ * extensions, such as my-component.extension.svelte, which will result in
```

Comment:
We could fix this by doing `name.split('.').pop()`. What do you think? Does this fail in cases I don't know yet?

Reply:
I'm not sure... honestly I don't think we can get it 100% right here, other than by recommending users to use `PascalCase` in their file names.

Maybe `name.split('.')[0]` is a good transformation because it will drop the extra extensions, which I doubt anybody will ever want in their class name...

Reply:
I agree. This would come after the `pascalCase`-transformation.

Reply:
`pascalCase` actually will transform `thing.thang` into `ThingThang`, so it needs to go before, I think.

Reply:
Since the comment is now outdated, could you update it, too?

Reply:
Done, I added some docs in the last commit. Those can be reviewed too, and then this is ready to go.